src/hpack/encoder.rs

```diff
@@ -384,10 +384,6 @@ fn encode_int<B: BufMut>(
 
     value -= low;
 
-    if value > 0x0fff_ffff {
```

Comment:
I feel like this is probably here for a reason... But I don't know what it is. At the least, I'd like to make sure the spec doesn't describe some upper limit that was being protected for here...

Reply:
I didn't find any limit in either HTTP/2 or HPACK, but I will check again.

Reply:
I checked a few other implementations:

[`hpack-rs`](https://github.com/mlalic/hpack-rs/blob/e833ecac324fb9457e04737f482328c3b5cc93fa/src/encoder.rs#L53-L120) doesn't have that hard limit.
[`httlib-hpack`](https://github.com/xpepermint/httlib-rs/blob/e733390c52fdc2d784864507877416705480db97/hpack/src/encoder/primitives.rs#L3-L70) also doesn't have that hard limit.
[`tokio-http2`](https://github.com/lambdastackio/tokio-http2/blob/fad8751ada3568d8cf4890d6f39ab3ee7d7c7301/src/hpack/encoder.rs#L53-L120) does the same as `hpack-rs`.
[`hpack-codec`](https://github.com/sile/hpack_codec/blob/cfc49c78d16dc38d9529ed20a380355d9c4c34ba/src/literal.rs#L11-L33) only supports encoding `u16` values.

Between this and our chat yesterday on Discord, IMO we should just merge this PR.